alttprbot/alttprgen/smvaria.py

```diff
@@ -9,6 +9,123 @@ async def generate_preset(settings, skills, race=False):
+        raise ValueError("Invalid start setting.  Must be DeepStart, RandomStart, NotDeepStart, ShallowStart or VanillaStart.")
+
+    splitDict = {
+        "FullCountdown": "FullWithHUD",
+        "M/m": "Major",
+        "RandomSplit": "random"
+    }
+
+    difficultyDict = {
+        "HarderDifficulty": "harder",
+        "HardDifficulty": "hard",
+        "MediumDifficulty": "medium",
+        "EasyDifficulty": "easy",
+        "BasicDifficulty": "easy"
+    }
+
+    morphDict = {
+        "LateMorph": "late",
+        "RandomMorph": "random",
+        "EarlyMorph": "early"
+    }
+
+    startDict = {
+        "DeepStart": [
+            'Aqueduct',
+            'Bubble Mountain',
+            'Firefleas Top'
+        ],
+        "RandomStart": [
+            'Aqueduct',
+            'Big Pink',
+            'Bubble Mountain',
+            'Business Center',
+            'Etecoons Supers',
+            'Firefleas Top',
+            'Gauntlet Top',
+            'Golden Four',
+            'Green Brinstar Elevator',
+            'Red Brinstar Elevator',
+            'Wrecked Ship Main'
+        ],
+        "NotDeepStart": [
+            'Big Pink',
+            'Business Center',
+            'Etecoons Supers',
+            'Gauntlet Top',
+            'Golden Four',
+            'Green Brinstar Elevator',
+            'Red Brinstar Elevator',
+            'Wrecked Ship Main'
+        ],
+        "ShallowStart": [
+            'Big Pink',
+            'Gauntlet Top',
+            'Green Brinstar Elevator',
+            'Wrecked Ship Main'
+        ],
+        "VanillaStart": [
+            'Landing Site'
+        ]
+    }
+
+    settings = {
+        "hud": "on",
+        "suitsRestriction": "off",
+        "variaTweaks": "on",
+
+        "majorsSplit": splitDict[split],
+        "majorsSplitMultiSelect": ['FullWithHUD', 'Major'],
+
+        "areaRandomization": "off" if area == "VanillaArea" else "on",
+        "areaLayout": "off" if area == "VanillaArea" else "on",
+        "lightAreaRandomization": "on" if area == "LightArea" else "off",
+
+        "bossRandomization": "off" if boss == "VanillaBoss" else "on",
+
+        "maxDifficulty": difficultyDict[difficulty],
+
+        "escapeRando": "off" if escape == "VanillaEscape" else "on",
+
+        "morphPlacement": morphDict[morph],
+
+        "startLocation": "Landing Site" if start == "VanillaStart" else "random",
+        "startLocationMultiSelect": startDict[start]
+    }
+
+    await ctx.send_message("Generating game, please wait.  If nothing happens after a minute, contact Synack.")
+    seed = await SuperMetroidVariaDiscord.create(
+        settings_preset="Season_Races",
+        skills_preset="newbie" if difficulty == "BasicDifficulty" else "Season_Races",
+        race=race,
+        settings_dict=settings
+    )
+    if not hasattr(seed, 'guid'):
+        raise Exception("Error: %s" % seed.data)
+    return seed
+
```

Comment:
Largely copied from the Choozo discord bot so I expect it will work.  Not sure how to test it though.  I do know it at least "compiles" (python doesn't error out on a syntax mistake)